analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -24,43 +24,22 @@
 
 namespace SonarAnalyzer.AnalysisContext;
 
-public class SonarAnalysisContextBase
+public abstract class SonarAnalysisContextBase
```

Comment:
None of these changes are needed to change the UtilityAnalyzerBase. It should all be reverted to the original state.

Reply:
It is needed for the UTs to compile:
https://github.com/SonarSource/sonar-dotnet/pull/6576/files#diff-8f80eb3e3e0e9cc4b126e1ab5dfcfafae31c5a7468e229ec475cb37af39bc8f4R143-R154

Reply:
No, it's not. Rewrite the UTs to make them work without that unneeded inheritance. You can take a look at tests from `AnalysisContext` directory to see how to properly scaffold the other stuff.